src/plasma/plasma_manager.cc

```diff
@@ -904,14 +921,14 @@ void request_transfer_from(PlasmaManagerState *manager_state,
     transfer_request->type = MessageType_PlasmaDataRequest;
     transfer_request->object_id = fetch_req->object_id;
 
-    if (manager_conn->transfer_queue.size() == 0) {
+    if (manager_conn->data_transfer_queue.size() == 0 && manager_conn->data_transfer_queue.size() == 0) {
```

Comment:
I think the same condition is repeated twice in this `if`; it should probably be
```
 if (manager_conn->data_transfer_queue.size() == 0 && manager_conn->data_request_queue.size() == 0) {
```
instead

Reply:
nice catch :)